generator_process.py

```diff
@@ -281,57 +348,101 @@ def update(self, n=1):
+            send_info("Starting")
+            
+            try:
+                tmp_stderr = sys.stderr = StringIO() # prompt2image writes exceptions straight to stderr, intercepting
+                generator.prompt2image(
+                    # a function or method that will be called each step
+                    step_callback=view_step,
+                    # a function or method that will be called each time an image is generated
+                    image_callback=image_writer,
+                    **args
+                )
+                if tmp_stderr.tell() > 0:
+                    tmp_stderr.seek(0)
+                    s = tmp_stderr.read()
+                    i = s.find("Traceback") # progress also gets printed to stderr so check for an actual exception
+                    if i != -1:
+                        s = s[i:]
+                        import re
+                        low_ram = re.search(r"(Not enough memory, use lower resolution)( \(max approx. \d+x\d+\))",s,re.IGNORECASE)
+                        if low_ram:
+                            send_exception(False, f"{low_ram[1]}{' or disable full precision' if args['precision'] == 'float32' else ''}{low_ram[2]}", s)
+                        elif s.find("CUDA out of memory. Tried to allocate") != -1:
+                            send_exception(False, f"Not enough memory, use lower resolution{' or disable full precision' if args['precision'] == 'float32' else ''}", s)
+                        else:
+                            send_exception(True, msg=None, trace=s) # consider all unknown exceptions to be fatal so the generator process is fully restarted next time
+                            return
+            except:
+                send_exception()
+                return
+            finally:
+                sys.stderr = stderr
+        elif intent == Intent.UPSCALE:
+            tmp_stderr = sys.stderr = StringIO()
+            json_len = int.from_bytes(stdin.read(8),sys.byteorder,signed=False)
+            if json_len == 0:
+                return # stdin closed
+            args = json.loads(stdin.read(json_len))
+            send_info("Starting")
             try:
-                generator = Generate(
-                    conf=models_config,
+                from absolute_path import REAL_ESRGAN_WEIGHTS_PATH
+                import cv2
+                from realesrgan import RealESRGANer
+                from realesrgan.archs.srvgg_arch import SRVGGNetCompact
+                # image = Image.open(args['input'])
+                image = cv2.imread(args['input'], cv2.IMREAD_UNCHANGED)
+                model = SRVGGNetCompact(num_in_ch=3, num_out_ch=3, num_feat=64, num_conv=32, upscale=4, act_type='prelu')
+                netscale = 4
+                send_info("Loading Upsampler")
+                upsampler = RealESRGANer(
+                    scale=netscale,
+                    model_path=REAL_ESRGAN_WEIGHTS_PATH,
+                    dni_weight=1,
```

Comment:
TypeError: RealESRGANer.__init__() got an unexpected keyword argument 'dni_weight'

Reply:
Ah, I installed 0.3.0 locally, but set the windows requirements to 0.2.5.0.